File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        sumk = sum(nums[:k])
        max_sum = sumk

        for i in range(k,len(nums)):
            
            sumk = sumk - nums[i-k] + nums[i]
            max_sum = max(max_sum,sumk)

        return max_sum/k

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {

        int maxSum = Integer.MIN_VALUE;
        for(int i =0;i<=nums.length-k;i++){

            int tempSum = sum(i,nums,k);
            maxSum = Math.max(maxSum,tempSum);

        }
        System.out.println(maxSum);
        return (double) maxSum/k;
        
    }

    public int sum(int start,int[] nums,int k){

        int sum = 0;
        for(int i=start;i<start+k;i++){
            sum = sum + nums[i];
        }
        return sum;
    }
}